compiler/x/codegen/OutlinedInstructions.hpp

```diff
@@ -145,4 +150,15 @@ class TR_OutlinedInstructions
    void setRematerializeVMThread() { _rematerializeVMThread = true; }
 
    };
+
+class TR_OutlinedGenerator
```

Comment:
This looks like a useful utility which I'd definitely want to make use of in the Z codegen. Can this be moved to a common location and Doxygen documented?

Reply:
Good suggestion. I've added the documents.
I also would like to make it a common utility; however, I prefer to keep it inside X scope for now. Its implementation is closely tied to X's ```TR_OutlinedInstructions```, and currently contains workaround that avoiding breaking downstream projects in the intermediate stage. For example using the deprecated constructor for now. Once downstream projects adopt it and the deprecated constructor is removed, ```TR_OutlinedGenerator``` will reach to its final stage, and we can move it to a common location.
I'm expecting another 1-2 pull requests to clean up deprecated code.